### PR TITLE
Create symlink to python binary instead of copying

### LINKED ISSUE #8: `python.app` version should match `python` version
Issue:
We keep running into issues on macOS with multiprocessing because the Python versions in `python.app` and `python` don't match. This causes segfaults in multiprocessing pools at least for the current `python`/`python.app` combination for Python 3.7 on `conda-forge`.

Example for `python[.app]` with Python 3.7:
```
$ python --version
Python 3.7.6
$ python.app --version
Python 3.7.3
```

@cbrnr had reported this issue to the Anaconda issue tracker, and it has since been fixed in `defaults`: https://github.com/ContinuumIO/anaconda-issues/issues/11633

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes GH-8

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hoechenberger** [comment]
@conda-forge-admin, please rerender

**Comment by hoechenberger** [comment]
@conda-forge-admin, please rerender

**Comment by hoechenberger** [comment]
@ChrisBarker-NOAA This is ready for review / I can merge if you're happy

@cbrnr et voilà, symlinked Python!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ package:
   version: 1.2
```
Comment (ChrisBarker-NOAA) [inline recipe/meta.yaml]: Since this is a change to how the Bundle is put together, it should probably be a bump to the version number:

Maybe 1.3.0 ?

**Comment by ChrisBarker-NOAA** [comment]
OK:

I set this up on my machine, and built it and tested it -- it seems to work, at least with wxPython and:

```
python                    3.8.2           hdc38147_4_cpython    conda-forge
python.app                1.2             py37h1de35cc_1204    local
wxpython                  4.0.7.post2      py38h1cd7e00_0    conda-forge
```
On MacOS v. 10.13.6

So:

- [ ] Bump the version  to 1.3
- [ ] Reset Build Number to 0

And I guess we are OK.

However, this still makes me nervous. This is weird stuff, and we're remaking it in a way that seems simpler -- there was probably a reason not to do that the first time.

I'd be more comfortable if there was more review from someone that "gets" this stuff.

Baring that -- at least some more testing -- could someone(s) please report the restults of testing this with a few more variants:

- [ ] MacOS v 10.12
- [ ] MacOS v 10.14

Other GUI Toolkits:

- [ ] TkInter
- [ ] PyQT
- [ ] PySide (If different)
- [ ] PyObjC

It's probably not practical to test the whole matrix, but a few more would be nice ...

**Comment by hoechenberger** [comment]
> This is weird stuff, and we're remaking it in a way that seems simpler -- there was probably a reason not to do that the first time.

The feedstock sources were essentially copied from Anaconda Inc.'s [legacy recipe](https://github.com/ContinuumIO/anaconda-recipes/tree/master/python.app). I suspect the original idea of the authors was to create `python.app` as an entirely self-contained `python` alternative, and that's why they decided to copy the binary and not to create a link.



> testing this with a few more variants:
> 
> * [ ]  MacOS v 10.12
> * [ ]  MacOS v 10.14

I don't have access to machines running these systems…

> Other GUI Toolkits:
> 
> * [ ]  TkInter
> * [ ]  PyQT
> * [ ]  PySide (If different)
> * [ ]  PyObjC

This, OTOH, sounds feasible.

**Comment by hoechenberger** [comment]
> This is weird stuff

To be honest, this entire `pythonw` thing is maximally weird and annoying, and it's so frustrating to see we still have to deal with this stuff on the `conda` side while the official Python.org builds had abandoned this _many_ years ago. (Sorry for the rant, it's just _really_ annoying and causes nothing but trouble!)

**Comment by hoechenberger** [comment]
FWIW, symlinks inside app bundles are used by other people too, e.g.:
https://stackoverflow.com/questions/27871099/creating-symlinks-in-osx-frameworks-inside-app-bundle and https://cutecoder.org/programming/code-signing-failure-due-to-symlink-folders/

**Comment by hoechenberger** [comment]
> Other GUI Toolkits:
> 
> * [x]  TkInter
> * [x]  PyQT
> * [x]  PySide (If different)
> * [x]  PyObjC

They all work for me on Catalina. Please find the test scripts I used below.

### TkInter
```python
from tkinter import Tk

win = Tk()
win.title("Title")
win.mainloop()
```

### PyQt 5
```python
import sys
from PyQt5.QtWidgets import QApplication, QWidget

if __name__ == '__main__':
    app = QApplication(sys.argv)

    win = QWidget()
    win.resize(250, 150)
    win.setWindowTitle('Simple')
    win.show()
    
    sys.exit(app.exec_())
```

### PySide 2
```python
import sys
from PySide2.QtWidgets import QApplication, QWidget

if __name__ == '__main__':
    app = QApplication(sys.argv)

    win = QWidget()
    win.resize(250, 150)
    win.setWindowTitle('Simple')
    win.show()
    
    sys.exit(app.exec_())
```

### PyObjC-Core
```python
import objc
from Foundation import *
from AppKit import *
from PyObjCTools import AppHelper


def main():
    app = NSApplication.sharedApplication()
    win = NSWindow.alloc()
    frame = ((200.0, 300.0), (250.0, 100.0))
    win.initWithContentRect_styleMask_backing_defer_(frame, 15, 2, 0)
    win.setTitle_("HelloWorld")

    quit = NSButton.alloc().initWithFrame_(((100.0, 10.0), (80.0, 80.0)))
    win.contentView().addSubview_(quit)
    quit.setTarget_(app)
    quit.setAction_("stop:")
    quit.setEnabled_(1)
    quit.setTitle_("Quit")

    win.display()
    win.orderFrontRegardless()
    AppHelper.runEventLoop()


if __name__ == "__main__":
    main()
```

### Output of `conda list`
```
# packages in environment at /Users/hoechenberger/miniconda3/envs/python.app:
#
# Name                    Version                   Build  Channel
beautifulsoup4            4.8.2                    py37_0    conda-forge
bzip2                     1.0.8                h0b31af3_2    conda-forge
ca-certificates           2019.11.28           hecc5488_0    conda-forge
certifi                   2019.11.28               py37_0    conda-forge
cffi                      1.14.0           py37h356ff06_0    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
conda                     4.8.2                    py37_0    conda-forge
conda-build               3.18.12                  py37_1    conda-forge
conda-package-handling    1.6.0            py37h0b31af3_1    conda-forge
cryptography              2.8              py37hafa8578_1    conda-forge
dbus                      1.13.6               h2f22bb5_0    conda-forge
expat                     2.2.9                h4a8c4bd_2    conda-forge
filelock                  3.0.10                     py_0    conda-forge
gettext                   0.19.8.1          h46ab8bc_1002    conda-forge
glib                      2.58.3          py37h577aef8_1002    conda-forge
glob2                     0.7                        py_0    conda-forge
icu                       64.2                 h6de7cb9_1    conda-forge
idna                      2.9                        py_1    conda-forge
jinja2                    2.11.1                     py_0    conda-forge
jpeg                      9c                h1de35cc_1001    conda-forge
krb5                      1.16.4               h1752a42_0    conda-forge
libarchive                3.3.3             h8912c15_1007    conda-forge
libclang                  9.0.1           default_hf57f61e_0    conda-forge
libcxx                    9.0.1                         1    conda-forge
libedit                   3.1.20170329      hcfe32e1_1001    conda-forge
libffi                    3.2.1             h6de7cb9_1006    conda-forge
libiconv                  1.15              h01d97ff_1005    conda-forge
liblief                   0.9.0                h3e78482_1    conda-forge
libllvm9                  9.0.1                ha1b3eb9_0    conda-forge
libpng                    1.6.37               h2573ce8_0    conda-forge
libpq                     12.2                 h554dc5a_0    conda-forge
libxml2                   2.9.10               h53d96d6_0    conda-forge
libxslt                   1.1.33               h320ff13_0    conda-forge
lz4-c                     1.8.3             h6de7cb9_1001    conda-forge
lzo                       2.10              h1de35cc_1000    conda-forge
markupsafe                1.1.1            py37h0b31af3_0    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
nspr                      4.20              h0a44026_1000    conda-forge
nss                       3.47                 hc0980d9_0    conda-forge
openssl                   1.1.1d               h0b31af3_0    conda-forge
pcre                      8.44                 h4a8c4bd_0    conda-forge
pip                       20.0.2                     py_2    conda-forge
pkginfo                   1.5.0.1                    py_0    conda-forge
psutil                    5.7.0            py37h0b31af3_0    conda-forge
py-lief                   0.9.0            py37h0ceac7d_1    conda-forge
pycosat                   0.6.3           py37h0b31af3_1002    conda-forge
pycparser                 2.20                       py_0    conda-forge
pyobjc-core               6.1                      py37_0    conda-forge
pyobjc-framework-cocoa    6.1                      py37_0    conda-forge
pyopenssl                 19.1.0                     py_1    conda-forge
pyqt                      5.12.3           py37he22c54c_1    conda-forge
pyqt5-sip                 4.19.18                  pypi_0    pypi
pyqtwebengine             5.12.1                   pypi_0    pypi
pyside2                   5.13.2           py37ha8f7116_2    conda-forge
pysocks                   1.7.1                    py37_0    conda-forge
python                    3.7.6           hfa4aa89_4_cpython    conda-forge
python-libarchive-c       2.9                      py37_0    conda-forge
python.app                1.2             py37h1de35cc_1204    file:///Users/hoechenberger/miniconda3/envs/python.app/conda-bld
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2019.3                     py_0    conda-forge
pyyaml                    5.3              py37h0b31af3_0    conda-forge
qt                        5.12.5               h514805e_3    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
requests                  2.23.0                   py37_0    conda-forge
ripgrep                   11.0.2               h01d97ff_3    conda-forge
ruamel_yaml               0.15.80         py37h0b31af3_1000    conda-forge
setuptools                45.3.0                   py37_0    conda-forge
six                       1.14.0                   py37_0    conda-forge
soupsieve                 1.9.4                    py37_0    conda-forge
sqlite                    3.30.1               h93121df_0    conda-forge
tk                        8.6.10               hbbe82c9_0    conda-forge
tqdm                      4.43.0                     py_0    conda-forge
urllib3                   1.25.7                   py37_0    conda-forge
wheel                     0.34.2                     py_1    conda-forge
xz                        5.2.4             h1de35cc_1001    conda-forge
yaml                      0.2.2                h0b31af3_1    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge
zstd                      1.4.4                he7fca8b_1    conda-forge
```

**Comment by ChrisBarker-NOAA** [comment]
OK then -- looking good.

Thanks for all your work on this, it should make things a tad easier.

@cpbotha: I'd be more comfortable merging if you chimed in as well.

**Comment by cpbotha** [comment]
Hi there everyone!

I reviewed the changes carefully, and they look good to me.

However, I do not use macOS as my primary dev platform anymore, so I don't have anything to practically test this on.

In spite of that, the PR has my modest vote.

**Comment by ChrisBarker-NOAA** [comment]
Well, we've tested with two versions of teh OS and 6 GUI toolkits. That's as good as it's going to get.

Thanks all!

**Pull request merged**